src/stash.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2010-2022 The pygit2 contributors
+ *
+ * This file is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License, version 2,
+ * as published by the Free Software Foundation.
+ *
+ * In addition to the permissions in the GNU General Public License,
+ * the authors give you unlimited permission to link the compiled
+ * version of this file into combinations with other programs,
+ * and to distribute those combinations without any restriction
+ * coming from the use of this file.  (The General Public License
+ * restrictions do apply in other respects; for example, they cover
+ * modification of the file, and distribution when not linked into
+ * a combined executable.)
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include "object.h"
+#include "error.h"
+#include "types.h"
+#include "utils.h"
+#include "oid.h"
+
+PyTypeObject StashType;
+
+
+PyDoc_STRVAR(Stash_oid__doc__, "Stashed state.");
+
+PyObject *
+Stash_oid__get__(Stash *self)
+{
+    if (!self->oid)
+        Py_RETURN_NONE;
```

Comment:
I don't think this can be `NULL`. If it's then there was an error when it was set, and it should be handled there.
So please remove testing whether oid is `NULL` through the code. And check whether `git_oid_to_python` returns `NULL` in `foreach_stash_cb`, if it does then there has been an error.